src/javascript/box.js

```diff
@@ -0,0 +1,108 @@
+import API from './api.js';
+
+const api = new API();
+const numberWithCommas = s => String(s).replace(/\B(?=(\d{3})+(?!\d))/g, ",");
+
+export default function Box({ target, buttonTarget, page = 1, count = 5 }) {
+  this.target = target;
+  this.buttonTarget = buttonTarget;
+  this.page = page;
+  this.count = count;
+  this.boxLength = 0;
+  this.boxPrefix = 'box__item';
+  this.buttonPrefix = 'more-button';
+  this.countTarget = null;
+}
+
+Box.prototype = {
+  init,
+  getBoxLength,
+  getBoxList,
+  renderBoxList,
+  getBoxBlock,
+  renderMoreButton,
+  isMoreItem,
+  setDisableMoreButton,
+  updateCount,
+  createMoreEvent,
+}
+
+Box.constructor = Box;
+
+async function init() {
+  this.boxLength = await this.getBoxLength();
+  this.renderMoreButton();
+  await this.renderBoxList();
```

Comment:
this.renderBoxList가 비동기 작업을 수행하는 것이 아니라 data-fetching하는 로직이 함께 수행되기 때문에 async함수가 되었네요. 두 가지 일을 분리하는 것이 다른 render메서드와의 일관성을 가지게 해서 좋을 것 같아요.

```suggestion
  const boxList = await api.getItem({ type: 'box', page: this.page, count: this.count});
  this.renderBoxList(boxList);
```